File: problems/02_two_pointers/0125_valid_palindrome/solution.swift
// Problem: 125. Valid Palindrome
// Difficulty: Easy
// URL: https://leetcode.com/problems/valid-palindrome/
//
// 【解法演進】清理 + 反轉 → 雙指標 O(n)/O(1)（最佳）
// 【最佳解核心思路】左右指標相向，跳過非英數，比較小寫字元。
// 【複雜度】時間：O(n)  空間：O(1)

import Foundation

class Solution {
    func isPalindrome(_ s: String) -> Bool {
        // TODO: implement
        return false
    }

    func isPalindrome_brute(_ s: String) -> Bool {
        // TODO: implement
        return false
    }
}

func checkBool(_ actual: Bool, _ expected: Bool, _ name: String) {
    if actual == expected { print("✓ \(name)") }
    else { print("✗ \(name): expected \(expected) but got \(actual)") }
}

// MARK: - Tests
// let s = Solution()

// Test 1
// let result = s.solve(...)

print("All tests passed!")

File: problems/02_two_pointers/0125_valid_palindrome/solution.py
"""
Problem: 125. Valid Palindrome
Difficulty: Easy
URL: https://leetcode.com/problems/valid-palindrome/

【解法演進】清理 + 反轉比較 O(n)/O(n) → 雙指標原地比較 O(n)/O(1)（最佳）
【最佳解核心思路】
    左右指標相向而行，跳過非英數字元，逐對比較（小寫不敏感）。
【複雜度】時間：O(n)  空間：O(1)
"""

class Solution:
    # 最佳解：雙指標
    def isPalindrome(self, s: str) -> bool:
        pass

    def isPalindrome_brute(self, s: str) -> bool:
        pass


if __name__ == "__main__":
    solution = Solution()

    # Test 1
    # result = solution.solve(...)
    # assert result == expected

    print("All tests passed!")
